File: spring_java_orm_sample/src/main/java/com/example/demo/data/domain/BookStockMovementSourceType.java
package com.example.demo.data.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.seasar.doma.Domain;

@Domain(valueType = int.class, factoryMethod = "of")
public enum BookStockMovementSourceType {
    PURCHASE_INVOICE(1),
    SALES_ORDER(2),
    STOCK_ADJUSTMENT(3),
    STORE_TRANSFER(4);

    private final int value;

    BookStockMovementSourceType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    @JsonCreator
    public static BookStockMovementSourceType of(int value) {
        for (BookStockMovementSourceType enm : values()) {
            if (enm.value == value) {
                return enm;
            }
        }
        throw new IllegalArgumentException("Unknown BookStockMovementSourceType value: " + value);
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/data/domain/BookStockMovementSourceType.kt
package com.example.demo.data.domain

import com.fasterxml.jackson.annotation.JsonCreator
import org.seasar.doma.Domain

@Domain(valueType = Int::class, factoryMethod = "of")
enum class BookStockMovementSourceType(val value: Int) {
    PURCHASE_INVOICE(1),
    SALES_ORDER(2),
    STOCK_ADJUSTMENT(3),
    STORE_TRANSFER(4);

    companion object {
        @JvmStatic
        @JsonCreator
        fun of(value: Int): BookStockMovementSourceType {
            for (enm in entries) {
                if (enm.value == value) {
                    return enm
                }
            }
            throw IllegalArgumentException("Unknown BookStockMovementSourceType value: " + value)
        }
    }
}
